cloudflare/resource_cloudflare_worker_secret.go

```diff
@@ -0,0 +1,119 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudflareWorkerSecret() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareWorkerSecretCreate,
+		Read:   resourceCloudflareWorkerSecretRead,
+		Delete: resourceCloudflareWorkerSecretDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareWorkersSecretImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"script_name": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: "The name of the Worker script to associate the secret with.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: "The name of the Worker secret.",
+			},
+			"secret_text": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Sensitive:   true,
+				Description: "The text of the Worker secret, this cannot be read back after creation and is stored encrypted .",
+			},
+		},
+	}
+}
+
+func resourceCloudflareWorkerSecretRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	scriptName := d.Get("script_name").(string)
+	name := d.Get("name").(string)
+
+	secrets, err := client.ListWorkersSecrets(context.Background(), scriptName)
```

Comment:
depending on what we are carrying here, `context.TODO` may be more appropriate. do you recall what the underlying library is wanting here?

Reply:
I don't understand. The library is listing the worker secrets by issuing a get request against the server.

Reply:
`context.TODO()` and `context.Background()` [share the same underlying functionality](https://github.com/golang/go/blob/master/src/context/context.go#L204-L218) however the way I read the intended usage is that `TODO()` is when we aren't explicitly expecting to use the context for deadlines, etc whereas `Background()` is when are explicitly expecting to use it in that way. My question is mainly around are we intentionally doing this or is it to just satisfy the function signature? In which case, `context.TODO()` may make more sense here. I'm not fussed too much  but do think we want to clarify.

Reply:
AFAICT context.TODO() is used when there is intent in the codebase to _eventually_ provide a background in future work. That is not the case here as we won't be passing a Context into this function in the future, so I'm happy with it as is :)